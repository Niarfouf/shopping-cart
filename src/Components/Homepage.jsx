import { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
const refArray = [
    [0, 'electronics'],
    [1, 'jewelery'],
    [2, "men's clothing"],
    [3, "women's clothing"],
]
export default function Homepage() {
    const [active, setActive] = useState(0)
    function left() {
        setActive((active) => (active === 0 ? 3 : active - 1))
    }
    function right() {
        setActive((active) => (active === 3 ? 0 : active + 1))
    }
    function capitalizeFirstLetter(string) {
        return string.charAt(0).toUpperCase() + string.slice(1)
    }
    useEffect(() => {
        const key = setInterval(() => {
            right()
        }, 3000)

        return () => {
            clearInterval(key)
        }
    }, [active])
    return (
        <>
            <div className={`caroussel image${active}`}>
                <div className="caroussel-buttons">
                    <button onClick={left} className="left-arrow">
                        <svg
                            xmlns="http://www.w3.org/2000/svg"
                            viewBox="0 0 24 24"
                        >
                            <title>menu-left</title>
                            <path d="M14,7L9,12L14,17V7Z" />
                        </svg>
                    </button>
                    <h1>{capitalizeFirstLetter(refArray[active][1])}</h1>
                    <button onClick={right} className="right-arrow">
                        <svg
                            xmlns="http://www.w3.org/2000/svg"
                            viewBox="0 0 24 24"
                        >
                            <title>menu-right</title>
                            <path d="M10,17L15,12L10,7V17Z" />
                        </svg>
                    </button>
                </div>

                <Link
                    className="shop-link"
                    to={{
                        pathname: `shop/${refArray[active][1]}`,
                    }}
                >
                    Start shopping
                </Link>

                <div className="caroussel-info">
                    {refArray.map((ref) => {
                        return (
                            <div
                                key={ref}
                                className={
                                    active === ref[0] ? 'dot filled' : 'dot'
                                }
                                onClick={() => setActive(ref[0])}
                            ></div>
                        )
                    })}
                </div>
            </div>
        </>
    )
}
