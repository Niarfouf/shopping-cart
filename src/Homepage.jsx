export default function Homepage() {
    return (
        <>
            <h1>test home</h1>
        </>
    )
}
