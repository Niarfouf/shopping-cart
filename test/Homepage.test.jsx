import { render, act, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Homepage from '../src/Components/Homepage'
import { MemoryRouter } from 'react-router-dom'

describe('Homepage', () => {
    it('render page snapshot', () => {
        const { container } = render(
            <MemoryRouter>
                <Homepage />
            </MemoryRouter>
        )

        expect(container).toMatchSnapshot()
    })
    it('left arrow button works', async () => {
        const user = userEvent.setup()
        render(
            <MemoryRouter>
                <Homepage />
            </MemoryRouter>
        )
        expect(screen.getByRole('heading').textContent).toMatch(/Electronics/i)
        const buttons = screen.getAllByRole('button')
        await user.click(buttons[0])
        expect(screen.getByRole('heading').textContent).toMatch(
            /Women's clothing/i
        )
    })
    it('right arrow button works', async () => {
        const user = userEvent.setup()
        render(
            <MemoryRouter>
                <Homepage />
            </MemoryRouter>
        )
        expect(screen.getByRole('heading').textContent).toMatch(/Electronics/i)
        const buttons = screen.getAllByRole('button')
        await user.click(buttons[1])
        expect(screen.getByRole('heading').textContent).toMatch(/Jewelery/i)
    })
    it('render heading and link and rotate every 3s', async () => {
        render(
            <MemoryRouter>
                <Homepage />
            </MemoryRouter>
        )

        expect(screen.getByRole('heading').textContent).toMatch(/Electronics/i)
        expect(screen.getByRole('link').textContent).toMatch(/Start shopping/i)
        await waitFor(
            () => {
                expect(screen.getByRole('heading').textContent).toMatch(
                    /Jewelery/i
                )
            },
            { timeout: 3000 }
        )
        await waitFor(
            () => {
                expect(screen.getByRole('heading').textContent).toMatch(
                    /Men's clothing/i
                )
            },
            { timeout: 3000 }
        )
        await waitFor(
            () => {
                expect(screen.getByRole('heading').textContent).toMatch(
                    /Women's clothing/i
                )
            },
            { timeout: 3000 }
        )
    }, 12000)
})
