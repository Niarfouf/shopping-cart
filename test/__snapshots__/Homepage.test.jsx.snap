// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage > render page snapshot 1`] = `
<div>
  <div
    class="caroussel image0"
  >
    <div
      class="caroussel-buttons"
    >
      <button
        class="left-arrow"
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            menu-left
          </title>
          <path
            d="M14,7L9,12L14,17V7Z"
          />
        </svg>
      </button>
      <h1>
        Electronics
      </h1>
      <button
        class="right-arrow"
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            menu-right
          </title>
          <path
            d="M10,17L15,12L10,7V17Z"
          />
        </svg>
      </button>
    </div>
    <a
      class="shop-link"
      href="/shop/electronics"
    >
      Start shopping
    </a>
    <div
      class="caroussel-info"
    >
      <div
        class="dot filled"
      />
      <div
        class="dot"
      />
      <div
        class="dot"
      />
      <div
        class="dot"
      />
    </div>
  </div>
</div>
`;
